### PR TITLE
IGNITE-14195 Query engines configuration

### PR DESCRIPTION
**Pull request by alex-plekhanov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java [hunk 1]
```diff
@@ -176,6 +179,37 @@ public SqlConfiguration setValidationEnabled(boolean validationEnabled) {
...
 
+    /**
+     * Sets query engines configuration.
+     * <p>
+     * There are several engines to execute SQL queries can be configured. If configured more than one engine, exact
+     * engine to execute the query can be chosen in run-time by {@code queryEngine} JDBC connection property or by
+     * {@code QUERY_ENGINE('engineName')} SQL query hint. If no query engine is explicitly chosen, default query engine
+     * will be used (see {@link QueryEngineConfiguration#setDefault(boolean)}).
+     * </p>
+     * </p>
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java:190]:
```suggestion
     * </p>
     * <p>
```
I'm not sure if closing tags are ever needed.

File: modules/calcite/src/main/java/org/apache/ignite/calcite/CalciteQueryEngineConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Query engine configuration for Calcite-based query engine.
+ */
+@IgniteExperimental
+public class CalciteQueryEngineConfiguration implements QueryEngineConfigurationEx {
+    /** Query engine name. */
+    public static final String ENGINE_NAME = "calcite";
+
+    /** */
+    private volatile boolean isDflt;
```
Comment (AMashenkov) [inline modules/calcite/src/main/java/org/apache/ignite/calcite/CalciteQueryEngineConfiguration.java:34]: Why does the configuration contains a `volatile` field?

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java (hunk 1 above)
Comment (alex-plekhanov) [reply modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java]: Ok, I've removed the closing tag, we don't use it usually

File: modules/calcite/src/main/java/org/apache/ignite/calcite/CalciteQueryEngineConfiguration.java (hunk 2 above)
Comment (alex-plekhanov) [reply modules/calcite/src/main/java/org/apache/ignite/calcite/CalciteQueryEngineConfiguration.java]: Fixed

**Review by ivandasch — APPROVED** [review]

**Comment by alex-plekhanov** [comment]
Merged to sql-calcite

**Pull request closed without merging**